test/cql-pytest/test_permissions.py

```diff
@@ -418,3 +418,21 @@ def test_udf_permissions_no_args(cql):
                     grant(cql, 'SELECT', table, username)
                     check_enforced(cql, username, permission='EXECUTE', resource=f'function {keyspace}.{fun}()',
                             function=lambda: user_session.execute(f'SELECT {keyspace}.{fun}() FROM {table}'))
+
+# Test that the create permission can't be granted on a single function
```

Comment:
Nitpick for future consideration (not important enough to delay this patch): I really like it when test comments also explain *why* what is being tested should be in that specific way. In the commit message you had a good explanation (about not having permissions on not-yet-existing objects) that could be directly copied here. 
Even more importantly, it's useful to mention that this test "reproduces #13822". It's useful because if we suddenly see this test failing in some old branch, or something, we immediately will know what we forgot to backport.